src/fhirdefs/impliedExtensions.ts

```diff
@@ -0,0 +1,457 @@
+  // Mappings R4 -> R5
+  DeviceUseStatement: 'DeviceUsage',
+  MedicationStatement: 'MedicationUsage',
+  MedicinalProduct: 'MedicinalProductDefinition',
+  MedicinalProductAuthorization: 'RegulatedAuthorization',
+  MedicinalProductContraindication: 'ClinicalUseIssue',
+  MedicinalProductIndication: 'ClinicalUseIssue',
+  MedicinalProductIngredient: 'Ingredient',
+  MedicinalProductInteraction: 'ClinicalUseIssue',
+  MedicinalProductManufactured: 'ManufacturedItemDefinition',
+  MedicinalProductPackaged: 'PackagedProductDefinition',
+  MedicinalProductPharmaceutical: 'AdministrableProductDefinition',
+  MedicinalProductUndesirableEffect: 'ClinicalUseIssue',
+  SubstanceSpecification: 'SubstanceDefinition',
+  // Mappings R5 -> R4
+  AdministrableProductDefinition: 'MedicinalProductPharmaceutical',
+  CapabilityStatement2: 'CapabilityStatement',
+  DeviceUsage: 'DeviceUseStatement',
+  Ingredient: 'MedicinalProductIngredient',
+  ManufacturedItemDefinition: 'MedicinalProductManufactured',
+  MedicationUsage: 'MedicationUseStatement',
+  MedicinalProductDefinition: 'MedicinalProduct',
+  PackagedProductDefinition: 'MedicinalProductPackaged',
+  RegulatedAuthorization: 'MedicinalProductAuthorization',
+  SubstanceDefinition: 'SubstanceSpecification'
+};
+
+// It's sad to ignore any child, but really, we don't care about these particular
+// children of the original element when building complex extensions.  Sorry kids!
+const IGNORED_CHILDREN = ['id', 'extension', 'modifierExtension'];
+
+export function isImpliedExtension(url: string) {
+  return IMPLIED_EXTENSION_REGEX.test(url);
+}
+
+export function materializeImpliedExtension(url: string, defs: FHIRDefinitions): any {
+  const match = url.match(IMPLIED_EXTENSION_REGEX);
+  if (match == null) {
+    logger.error(
+      `Cannot materialize implied extension (${url}) since the URL does not match the implied extension pattern ` +
+        'http://hl7.org/fhir/[version]/StructureDefinition/extension-[Path]} with version 1.0, 3.0, 4.0, or 5.0.'
+    );
+    return;
+  }
+
+  const [, version, id, type] = match;
+  const supplementalPackage = VERSION_TO_PACKAGE_MAP[version]; // guaranteed not to be null thanks to REGEX check above
+  const supplementalDefs = defs.getSupplementalFHIRDefinitions(supplementalPackage);
+  if (supplementalDefs == null) {
+    const extRelease = version === '1.0' ? 'r2' : `r${version[0]}`;
+    const fhirVersion = defs.fishForFHIR('StructureDefinition', Type.Resource)?.fhirVersion;
+    logger.error(
+      `Cannot materialize implied extension: ${url}.\n` +
+        'To fix this, add the following dependency to your sushi-config.yaml file:\n' +
+        `  hl7.fhir.extensions.${extRelease}: ${fhirVersion}`
+    );
+    return;
+  }
+
+  const sd = supplementalDefs.fishForFHIR(type);
+  if (sd == null) {
+    logger.error(
+      `Cannot materialize implied extension (${url}) since ${type} is not a valid resource or data type in ${supplementalPackage}`
```

Comment:
We're not consistent about it at all. (I just checked). I thought that I was adding a period at the end if the error was multiple sentences, but leaving it off if it was just a phrase. But it seems that is not actually what I was doing. Oops. I can add periods.

And by the way, I appreciate the nitpickiness. I usually try to avoid inconsistencies like this -- so appreciate your help in pointing them out.

Reply:
Fixed in ce58754